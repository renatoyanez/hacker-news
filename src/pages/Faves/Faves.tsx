import React from 'react';
import { IHits } from 'interfaces';
import { NewsListSection } from 'components';

const Faves = () => {
	const faves = JSON.parse(
		localStorage.getItem('faves') as string,
	) as IHits[];

	return (
		<div>
			<NewsListSection news={faves} />
		</div>
	);
};

export default Faves;
