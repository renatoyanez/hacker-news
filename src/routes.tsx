import { ReactElement } from 'react';
import { Route, Routes as Switch } from 'react-router-dom';
import { Faves, Home } from 'pages';

type RouteListType = {
	HOME: string;
	FAVES: string;
};

export const RouteList: RouteListType = {
	HOME: '/',
	FAVES: '/my-faves',
};

const Routes = (): ReactElement => {
	return (
		<Switch>
			<Route path={RouteList.HOME} element={<Home />} />
			<Route path={RouteList.FAVES} element={<Faves />} />
		</Switch>
	);
};

export default Routes;
