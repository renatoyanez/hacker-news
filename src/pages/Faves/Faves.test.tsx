import React from 'react';
import { Router as MemRouter } from 'react-router-dom';
import { cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import { RouteList } from 'routes';
import Faves from './Faves';

afterEach(cleanup);

describe('Faves', () => {
	const history = createMemoryHistory({
		initialEntries: [RouteList.FAVES],
	});

	it('renders correctly', () => {
		const tree = create(
			<MemRouter location={history.location} navigator={history}>
				<Faves />
			</MemRouter>,
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
