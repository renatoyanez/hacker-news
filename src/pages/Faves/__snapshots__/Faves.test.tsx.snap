// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Faves renders correctly 1`] = `
<div>
  <div>
    <section
      className="cards"
    />
  </div>
</div>
`;
